src/connectors/hierarchical-menu/connectHierarchicalMenu.js

```diff
@@ -275,11 +275,8 @@ export default function connectHierarchicalMenu(renderFn, unmountFn = noop) {
         const path = searchParameters.getHierarchicalFacetBreadcrumb(
           hierarchicalFacetName
         );
-        if (!path || path.length === 0) return uiState;
-        if (
-          uiState.hierarchicalMenu &&
-          isEqual(path, uiState.hierarchicalMenu[hierarchicalFacetName])
-        ) {
+
+        if (!path.length) {
```

Comment:
The function [`getHierarchicalFacetBreadcrumb`](https://github.com/algolia/algoliasearch-helper-js/blob/94f40da97a0efb4845436f207ddbf5efca223514/src/SearchParameters/index.js#L1467-L1482)  always return an array.